src/api/v1/Auth/controller.js

```diff
@@ -1,22 +1,137 @@
+                const { dataValues } = await model.create({
+                    name: name,
+                    email: email,
+                    mobile: mobile,
+                    password: hashedPassword,
+                });
+
+                if (Object.keys(dataValues)?.length) {
+                    const token = await generateUserToken(dataValues, role);
+
+                    return sendSuccessResp(res, {
+                        status: 200,
+                        data: {
+                            isExist: 0,
+                            token,
+                            message: "Sign Up Successfully!",
+                        },
+                    });
+                }
+            } else {
+                return sendFailureResp(res, {
+                    status: 400,
+                    data: {
+                        isExist: 1,
+                        message: "User already exists",
+                    },
+                });
             }
-        })
+        } catch (err) {
+            return sendFailureResp(res, {
+                status: 500,
+                data: {
+                    message: "Something went wrong!",
+                },
+            });
+        }
     },
 
-    login: (req, res) => {
-        sendSuccessResp(res, {
-            data: {
-                message: "Maan Lo Ki User Login ho Gya",
-            }
-        })
+    login: async (req, res) => {
+        await sequelize
+            .sync()
+            .then(async () => {
+                const { role, email, password } = req.body;
+
+                const user = await findUser(role, email);
+
+                if (!user) {
+                    return sendFailureResp(res, {
+                        data: {
+                            message: "User not found",
+                        },
+                    });
+                }
+
+                const hashComparison = await compareHash(
+                    password,
+                    user?.dataValues?.password
+                );
+
+                if (!hashComparison) {
+                    return sendFailureResp(res, {
+                        data: {
+                            message: "Wrong Password",
+                        },
+                    });
+                }
+
+                const token = await generateUserToken(user?.dataValues, role);
+
+                return sendSuccessResp(res, {
+                    data: {
+                        message: "valid login deatils, Access Provided",
+                        token,
+                    },
+                });
+            })
+            .catch((error) => {
+                console.error("Unable to find User : ", error);
+                sendFailureResp(res, {
+                    data: {
+                        message: "something went wrong",
+                    },
+                });
+            });
+    },
+};
+
+async function generateUserToken(dataValues, role) {
```

Comment:
no need of separate function for token generation its already decoupled enough

Reply:
if Token must be generated on successful Signup and login,  a separate function is required